### ISSUE TITLE
[Bug] rl_with_openpipe_art: inference targets base model instead of LoRA adapter during GRPO training

### ISSUE THREAD

**Issue by aslanshi** [NVIDIA/NeMo-Agent-Toolkit#1649]
### Version

1.4.0

### Summary
The `rl_with_openpipe_art` cookbook's `config.yml` sets `openpipe_llm.model_name` to `Qwen/Qwen2.5-3B-Instruct`, but ART registers the LoRA adapter under `backend.name` (`tic_tac_toe_training_run`). Every epoch, NAT runs inference against the base model instead of the latest LoRA checkpoint, making GRPO training ineffective.

### How ART Routing Works
Per the ART client docs, inference must use `model.get_inference_name()` (the training run name), not the base model name, so that:
> "Each new LoRA instance will be automatically loaded onto the vLLM server... inference requests are always routed to the latest version of the model."

### The Bug
```
# config.yml (abbreviated)
llms:  
  openpipe_llm:    
     model_name: Qwen/Qwen2.5-3B-Instruct  # always hits base model

trainer_adapters:  
   openpipe_trainer_adapter:    
      backend:      
         name: "tic_tac_toe_training_run"      # LoRA adapter registered here      
         base_model: "Qwen/Qwen2.5-3B-Instruct"
```

Rewards are always computed from the unchanged base model. The RL loop cannot converge.

### Impact
- Training is a no-op: GRPO updates LoRA weights that are never queried for the next evaluation.
- Misleading to users: Stagnant reward curves appear to be a tuning problem rather than a config bug.

### Why Full-Weight Training Would Mask This Bug
The README states ART uses "TorchTune for GRPO weight updates," but the config specifies no `torchtune_args`, so ART falls back to Unsloth LoRA finetuning. With full-weight training, vLLM updates weights in-place under the same model name, so the mismatch wouldn't matter.

Note: In the current ART API (v0.5.11), `TorchtuneArgs` has been replaced by TinkerArgs. Full-weight training can alternatively be enabled through init_args.full_finetuning:
```
trainer_adapters:  
   openpipe_trainer_adapter:    
      backend:      
         init_args:        
            max_seq_length: 8192        
            full_finetuning: true  # enables full-weight training via Unsloth
```
The NAT plugin is pinned to ART 0.4.11 which still references `art.dev.TorchtuneArgs`. Regardless, the README should clarify that the current config uses LoRA by default, and explain the implications for model naming.

### Suggested Fix
1. `config.yml` and `config_post_train.yml` -- change model_name to match `backend.name`:
```
llms:  
   openpipe_llm:    
      model_name: tic_tac_toe_training_run  # must match backend.name for LoRA routing
```
2. README -- clarify that the default config uses Unsloth LoRA (not TorchTune full-weight), and that `model_name` must match `backend.name` when using LoRA.

### Affected Files
- examples/finetuning/rl_with_openpipe_art/src/rl_with_openpipe_art/configs/config.yml
- examples/finetuning/rl_with_openpipe_art/src/rl_with_openpipe_art/configs/config_post_train.yml
- examples/finetuning/rl_with_openpipe_art/README.md

### Minimum reproducible example

```shell

```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>

### Other/Misc.

Avg rewards per epoch before and after fixing the inference pointer;

Before:
<img width="1782" height="1030" alt="Image" src="https://github.com/user-attachments/assets/873831a7-6272-4447-b1fc-e5a4ece7b58f" />

After:
<img width="1781" height="1030" alt="Image" src="https://github.com/user-attachments/assets/a6d16935-25b6-40a7-85d5-034caa96ee56" />

Note: the big drop at the final epoch may be related to some vLLM float reporting issues, which lies more on the OpenPipe and/or vLLM side. Needs more investigation.

### Code of Conduct

- [x] I agree to follow the NeMo Agent Toolkit Code of Conduct
- [x] I have searched the [open bugs](https://github.com/NVIDIA/NeMo-Agent-Toolkit/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report
